lib/beaker/dsl/helpers/puppet_helpers.rb

```diff
@@ -504,13 +504,21 @@ def run_agent_on(host, arg='--no-daemonize --verbose --onetime --test',
         #                            or a role (String or Symbol) that identifies one or more hosts.
         # @param ip_spec [Hash{String=>String}] a hash containing the host to ip
         #   mappings
+        # @param aliases Array[String] an array containing the host alias to apply ip mappings
         # @example Stub puppetlabs.com on the master to 127.0.0.1
         #   stub_hosts_on(master, 'puppetlabs.com' => '127.0.0.1')
-        def stub_hosts_on(machine, ip_spec)
+        def stub_hosts_on(machine, ip_spec, aliases=[])
           block_on machine do | host |
             ip_spec.each do |address, ip|
+              manifest =<<-EOS.gsub /^\s+/, ""
+                host { '#{address}':
+                  \tensure       => present,
+                  \tip           => '#{ip}',
+                  \thost_aliases => #{aliases},
```

Comment:
Hmm. I am not clear on how to resolve this. If I am interpreting the declared interface in the doc correctly, it is only expecting a single host/ip map. Obviously, however the method supports a longer hash.
In order to map the aliases to a specific host in the hash we would need to change the signature of the hash argument. This would be a braking change to these methods.

Any suggestions?